AK/FixedArray.h

```diff
@@ -87,8 +87,17 @@ class FixedArray {
         other.m_size = 0;
         other.m_elements = nullptr;
     }
-    // NOTE: Nobody uses this function, so we just explicitly delete it.
-    FixedArray<T>& operator=(FixedArray<T>&&) = delete;
+    FixedArray<T>& operator=(FixedArray<T>&& other)
+    {
+        if (this == &other)
+            return *this;
+        clear();
+        m_size = other.m_size;
+        m_elements = other.m_elements;
+        other.m_size = 0;
+        other.m_elements = nullptr;
+        return *this;
+    }
```

Comment:
There's a reason why this doesn't exist: It deallocates. The entire point of FixedArray is that it's doing one allocation on initialization and one deallocation on destruction. This function violates that in a non-obvious way. Also, the clear() function is something I want to kill anyways (because it violates that principle as well) and it just doesn't seem right. If you want a dynamic array-like list, you use Vector. If you want a static array-like list, you use FixedArray. No in-betweens please :^)

Reply:
Completely correct! I'll remove this.